accurest-core/src/main/groovy/io/codearte/accurest/util/JsonToJsonPathsConverter.groovy

```diff
@@ -62,17 +62,25 @@ class JsonToJsonPathsConverter {
 			// JSON with a list of primitives ["a", "b", "c"] in root issue #266
 		} else if (key.isIteratingOverNamelessArray() && value instanceof List && listContainsOnlyPrimitives(value)) {
 			value.each {
-				traverseRecursively(Object, key.arrayField().contains(it), it, closure)
+				traverseRecursively(Object, key.arrayField().contains(ContentUtils.returnParsedObject(it)),
+						ContentUtils.returnParsedObject(it), closure)
+			}
+		// JSON containing list of primitives { "partners":[ { "role":"AGENT", "payment_methods":[ "BANK", "CASH" ]	} ]
+		} else if (value instanceof List && listContainsOnlyPrimitives(value)) {
```

Comment:
You have to treat JSON with array of primitives in another way than array with maps